src/supernode/src/test/java/com/alibaba/dragonfly/supernode/repository/TaskRepositoryTest.java

```diff
@@ -0,0 +1,148 @@
+package com.alibaba.dragonfly.supernode.repository;
+
+import com.alibaba.dragonfly.supernode.common.domain.Task;
+import com.alibaba.dragonfly.supernode.common.enumeration.CdnStatus;
+import com.alibaba.dragonfly.supernode.repository.TaskRepository;
+import com.diffblue.deeptestutils.Reflector;
+import org.junit.Assert;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.ExpectedException;
+
+import java.lang.reflect.InvocationTargetException;
+import java.lang.reflect.Method;
+
+public class TaskRepositoryTest {
+
+  @Rule public ExpectedException thrown = ExpectedException.none();
+
+  /* testedClasses: TaskRepository */
+
+  /*
+   * Test generated by Diffblue Deeptest.
+   * This test case covers:
+   *  - conditional line 90 branch to line 90
+   *  - conditional line 90 branch to line 91
+   */
+
+  @Test
+  public void computePieceSizeInputNegativeOutputPositive()
+      throws NoSuchMethodException, IllegalAccessException, InvocationTargetException {
+
+    // Arrange
+    final TaskRepository objectUnderTest = new TaskRepository();
+    final Long httpFileLen = -209_715_200L;
+
+    // Act
+    final Class<?> classUnderTest =
+        Reflector.forName("com.alibaba.dragonfly.supernode.repository.TaskRepository");
+    final Method methodUnderTest =
+        classUnderTest.getDeclaredMethod("computePieceSize", Reflector.forName("java.lang.Long"));
+    methodUnderTest.setAccessible(true);
+    final Integer retval = (Integer)methodUnderTest.invoke(objectUnderTest, httpFileLen);
+
+    // Assert result
+    Assert.assertEquals(new Integer(4_194_304), retval);
+  }
+
+  /*
+   * Test generated by Diffblue Deeptest.
+   * This test case covers:
+   *  - conditional line 90 branch to line 90
+   *  - conditional line 90 branch to line 93
+   *  - conditional line 95 branch to line 95
+   */
+
```

Comment:
@peterschrammel is this an acceptable case for reflection? Both myself and Peter have come to the conclusion that it is
Source:http://cx.dashboard.diffblue.com/Dragonfly/1/tests/5bb34abd9709611b51d91f4b/trace

Reply:
👍